<!-- 
 yolo_detector.launch

 Copyright (C) 2018 Alessandro Francescon
 
 This program is free software: you can redistribute it and/or modify
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation, either version 3 of the License.
 
 This program is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 GNU General Public License for more details.
 
 You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
-->

<launch>

	<!-- The nodelet manager -->
	<node pkg="nodelet" type="nodelet" name="image_nodelet" args="manager">
		<remap from="/cv_camera/image_raw" to="/camera/image" />
                <param name="device_id" value="/dev/video0" />
	</node>

	<!-- The camera nodelet -->
	<node pkg="nodelet" type="nodelet" name="cv_camera" args="load cv_camera/CvCameraNodelet image_nodelet">
	</node>

        <!-- The Yolo detector nodelet -->
        <node pkg="nodelet" type="nodelet" name="yolo_detector" args="load darknet_cv/YoloDetectorNodelet image_nodelet">
                <remap from="image" to="/camera/image" />
                <param name="config_file" value="yolov3-tiny.cfg" />
                <param name="weights_file" value="yolov3-tiny.weights" />
                <param name="names_file" value="coco.names" />
                <param name="threshold" value="0.5" />
                <param name="hier_threshold" value="0.5" />
                <param name="nms_threshold" value="0.45" />
	</node>

        <!-- The debug image viewer -->
        <node pkg="image_view" type="image_view" name="debug_image_viewer">
                <remap from="image" to="/image_debug" />
	</node>

        <!-- The rqt topic gui -->
        <node pkg="rqt_topic" type="rqt_topic" name="rqt_topic">
        </node>

</launch>
